packages/sdk/src/client.ts

```diff
@@ -677,6 +723,211 @@ export class IndexerClient {
+					// wait for the request to be delivered on Hyperbridge
+					while (!request || request.statuses.length < 2) {
+						await sleep(self.config.pollInterval)
+						request = await self.queryGetRequest(hash)
+					}
+
+					status =
+						request.dest === self.config.hyperbridge.stateMachineId
+							? RequestStatus.DESTINATION
+							: RequestStatus.HYPERBRIDGE_DELIVERED
+
+					yield {
+						status,
+						metadata: {
+							blockHash: request.statuses[1].metadata.blockHash,
+							blockNumber: request.statuses[1].metadata.blockNumber,
+							transactionHash: request.statuses[1].metadata.transactionHash,
+						},
+					}
+					break
+				}
+
+				// the request has been verified and aggregated on Hyperbridge
+				case RequestStatus.HYPERBRIDGE_DELIVERED: {
+					// Get the latest state machine update for hyperbridge on the destination chain
+					let hyperbridgeFinalized: StateMachineUpdate | undefined
+					let index = request.source === self.config.hyperbridge.stateMachineId ? 0 : 1
+					while (!hyperbridgeFinalized) {
+						await sleep(self.config.pollInterval)
+						hyperbridgeFinalized = await self.queryStateMachineUpdateByHeight({
+							statemachineId: self.config.hyperbridge.stateMachineId,
+							height: request.statuses[index].metadata.blockNumber,
+							chain: request.dest,
+						})
+					}
+
+					const destChain = await getChain(self.config.dest)
+					const hyperbridge = await getChain({
+						...self.config.hyperbridge,
+						hasher: "Keccak",
+					})
+
+					const proof = await hyperbridge.queryRequestsProof(
+						[getRequestCommitment(request)],
+						request.dest,
+						BigInt(hyperbridgeFinalized.height),
+					)
+
+					const calldata = destChain.encode({
+						kind: "GetRequest",
+						proof: {
+							stateMachine: self.config.hyperbridge.stateMachineId,
+							consensusStateId: self.config.hyperbridge.consensusStateId,
+							proof,
+							height: BigInt(hyperbridgeFinalized.height),
+						},
+						requests: [request],
+						signer: pad("0x"),
+					})
+
+					yield {
+						status: RequestStatus.HYPERBRIDGE_FINALIZED,
+						metadata: {
+							blockHash: hyperbridgeFinalized.blockHash,
+							blockNumber: hyperbridgeFinalized.height,
+							transactionHash: hyperbridgeFinalized.transactionHash,
+							calldata,
+						},
+					}
+					status = RequestStatus.HYPERBRIDGE_FINALIZED
+					break
+				}
+
+				// request has been finalized by hyperbridge
+				case RequestStatus.HYPERBRIDGE_FINALIZED: {
+					// wait for the request to be delivered to the destination
+					let delivered = request.statuses.find((s) => s.status === RequestStatus.DESTINATION)
+					while (!request || !delivered) {
+						await sleep(self.config.pollInterval)
+						request = await self.queryGetRequest(hash)
+						delivered = request?.statuses.find((s) => s.status === RequestStatus.DESTINATION)
+					}
+
+					let index = request.source === self.config.hyperbridge.stateMachineId ? 1 : 2
```

Comment:
same thing here, if it was from hyperbridge then the request is already completed